src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,51 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_CAPACITY = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[STORAGE_CAPACITY];
+        values = (V[]) new Object[STORAGE_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == STORAGE_CAPACITY) {
+            return;
+        }
+        int index = indexOf(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
+        int index = indexOf(key);
+        if (index != -1) {
```

Comment:
Minor comment

```suggestion
        return index == -1 ? null : values[index];
```

Let's create constant for -1 value